Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM python:3.6.5-jessie
+
+env DEBIAN_FRONTEND noninteractive
+RUN apt-get update && \
+    apt-get -y install \
+            dbus-x11 \
+            libdbus-1-dev \
+            libespeak1 \
+            libnotify-cil-dev \
+            notification-daemon \
+            sudo \
+            xorg
+
+RUN pip3 install virtualenv
+
+
+COPY . /app
+WORKDIR /app
+
+RUN sudo /bin/bash ./docker.sh
```

Comment:
Why use sudo ? Are you not still root user at this point ? 
In any way, best pratice is to reduce number of docker layers. Lines 20 to 24 could be added in single layer.

`RUN  /bin/bash ./docker.sh && \
    useradd -Um jarvis && \
    echo "jarvis  ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers && \
    chown -R jarvis /app 
` 

By the way since you add a line into /etc/sudoers you're definetely root at this point, remove previous sudo.